src/react-components/ui-root.js

```diff
@@ -638,36 +670,37 @@ class UIRoot extends Component {
     this.props.scene.emit("add_media", media);
   };
 
-  closeDialog = () => {
-    this.setState({ dialog: null });
+  closeDialog = success => {
+    if (this.state.dialog) {
+      this.setState({ dialog: null });
+    } else {
+      // If dialog was successful (eg user hit "OK") then move forward in history, o/w go back.
+      //
+      // This makes it so if you create an object, back will re-show the create object dialog,
+      // but if you cancel, it will not.
+      if (success) {
+        this.props.history.push("/");
```

Comment:
Again, not sure I really desire to hit back to re-open the dialog. I feel like if I am looking at a hubs room, hitting back should take me back to the lobby, if I am looking at a modal, it should close the modal.